### ISSUE TITLE
Calculation comes back as blank type when inside a list

### ISSUE THREAD

**Issue by jptaranto** [sass/node-sass#1637]
Within my typey sass library (https://github.com/jptaranto/typey) I have a map which allows people to define font-sizes and line-heights. The map supports shorthand as so:

```
$typestyles: (
  big: 30px 40/30,
  small: 16px 24/16,
);
```

The first number is the font-size, the second as a calculation that returns the ratio for the line-height. It's easy to specify it like that because that gives you original pixel value of the line-height, whilst converting it over to a unitless ratio. For instance 40/30 = 1.33333.

This works great with ruby sass, however with node-sass - using type-of($value) - (where $value is the line-height calculation) returns blank, i.e. "". 

If I wrap it in (), i.e. (40/30) - it works fine. So I guess node-sass is not calculating it at some stage in the process. Perhaps because the shorthand is a list, and I am using an @each to get the value.

If I do not use shorthand as so, it also works fine (in node-sass):

```
$typestyles: (
  big: (
    font-size: 30px,
    line-height: 40/30
  )
);
```

If you want to try out the code, it's all inside the typeset mixin, inside typey: 
https://github.com/jptaranto/typey/blob/master/stylesheets/typey/mixins/_typeset.scss

**Comment by xzyfer** [comment]
Sorry, please produce a copy-pasteable example of Sass code that produces this issue so I can debug locally. I personally don't have time to digging through a framework.

**Comment by jptaranto** [comment]
Hey of course, sorry about that. The code is remarkably simple. Just in case it's a versioning thing this is with node-sass 3.4.2

```sass
$list: 1 1/2;

@each $value in $list {
  @warn type-of($value);
}
```

That will show "number" for the first item, and blank for the second.

**Comment by saper** [comment]
Some random version of sassc produces thx:

````
WARNING: number
Backtrace:
        t.scss:7

WARNING: 
Backtrace:
        t.scss:7
````

so you'd expect that the second expression is a unit less number type?
I think it belongs to https://github.com/sass/libsass (our C++ back end) and it's worth checking if nobody raised it there.

**Comment by xzyfer** [comment]
This has been fixed in the latest LibSass. I will land in node-sass@v3.10.0

**Comment by nschonni** [comment]
Closing as this should be fixed in the 4.0 release since it updated libsass. Feel free to reopen if you still see the issue after updating
